### PR TITLE
add mac and Windows e2e test for odo

### PR DESCRIPTION
**Pull request by anandrkskd**
Signed-off-by: anandrkskd <anandrkskd@gmail.com>

This PR adds tests running on  MacOS and Windows for odo.

fixes  https://github.com/openshift/odo/issues/4408

### CODE REVIEW

File: ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml [hunk 1]
```diff
@@ -159,6 +159,112 @@ presubmits:
...
+    context: ci/prow/psi-v4.7-integration-e2e-test-macos
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/prowgen-controlled: "true"
+      ci-operator.openshift.io/variant: psi
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-odo-main-psi-v4.7-integration-e2e-test-macos
+    optional: true
```
Comment (prietyc123) [inline ci-operator/jobs/openshift/odo/openshift-odo-main-presubmits.yaml:177]: Could you please make both the jobs as required?

**Review by prietyc123 — CHANGES_REQUESTED** [review]

**Comment by anandrkskd** [comment]
Fixes openshift/odo#4412

**Pull request closed without merging**